New area "moderation"

annando opened the pull request:
All user and content based admin stuff is moved to the moderation area.

nupplaphil commented on src/Module/BaseModeration.php:
```diff
@@ -0,0 +1,110 @@
+ *
+ * @package Friendica\Module
+ */
+abstract class BaseModeration extends BaseModule
```
Please replace all `DI` calls with a `protected` variable, set inside the `__constructor()`

See the last merged PRs of @MrPetovan as a good blue print

annando replied:
What does "protected" mean?

Quix0r replied:
methods or class fields with `protected` can only be invoked from the class itself and inheriting classes (means `extends`).

nupplaphil commented on src/Module/BaseModeration.php:
```diff
@@ -0,0 +1,110 @@
+	 * @throws HTTPException\ForbiddenException
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	public static function checkModerationAccess(bool $interactive = false)
```
As a result of my last comment, please make this class non-`static` :-)

nupplaphil commented on src/Module/BaseModeration.php:
```diff
@@ -0,0 +1,110 @@
+			'$modurl'    => 'moderation/'
+		]);
+
+		return '';
```
If it doesn't return anything, shouldn't this method be a `rawContent()`?

What do you think @MrPetovan ?

MrPetovan replied:
No, this is meant as a common method for all the child class `content()` methods. It's a little clunky because we don't use the return value but that's how we've done it for other pages with a similar common display (like the admin and settings modules)

nupplaphil commented on src/Module/Moderation/Summary.php:
```diff
@@ -0,0 +1,69 @@
+{
+	protected function content(array $request = []): string
+	{
+		parent::content();
```
You can delete this since it's just a empty hull

annando replied:
What is an empty hull? Don't we call the parent class here which has got some functionality in that function?

MrPetovan replied:
Actually this call is checking credentials, so far from being an empty hull!

nupplaphil replied:
oups .. thanks for the hint @MrPetovan ..

nupplaphil commented on src/Module/Moderation/Summary.php:
```diff
@@ -0,0 +1,69 @@
+use Friendica\Core\Logger;
+use Friendica\Core\Renderer;
+use Friendica\Database\DBA;
+use Friendica\DI;
```
Same as above, please replace all calls with proper `protected $variable` usages so no `DI::` is left

nupplaphil commented:
This is a great idea, maybe we can add a new role `moderator` which can use this panel :-)

annando commented:
This PR is meant as a first step. This was just a simple 1:1 move of the user and content based activities. The next step is to extend and fix the functionality. There is at least one optical bug, but I haven't changed it to make the PR more easy to check. Then after all functionality is added, we can introduce that new role. See also here: https://forum.friendi.ca/display/ec054ce7-9163-64b6-d730-e5f725780117

MrPetovan commented:
@annando Do you need me to spring on this PR?

MrPetovan commented:
Wrong button 🤦‍♂️

annando commented:
@MrPetovan yeah, I don't want to spend too much time on this PR, since there is some more stuff to do, like some bug with quoted posts and a lot of API stuff.

MrPetovan commented:
No problem, leave it with me, I'll make the necessary changes.

MrPetovan committed:
New area "moderation"

- Moved several admin pages to the moderation area
- ACL still is checking for administrator credentials

MrPetovan committed:
Updated main translation file after moving strings

Quix0r reviewed (approved):
Type-hint check: OKAY

MrPetovan commented:
Technically I could merge this since I didn't open the PR, but with my work on top of @annando I feel uneasy about it. All the moved pages were tested open.
